src/error.rs

```diff
@@ -53,6 +54,8 @@ impl<'tcx> Error for EvalError<'tcx> {
         match *self {
             EvalError::FunctionPointerTyMismatch(..) =>
                 "tried to call a function through a function pointer of a different type",
+            EvalError::ZstPointerDeref =>
+                "tried to deref a pointer to the ZST allocation",
```

Comment:
This is confusing. Dereferences are semantic representations of memory locations, they don't *actually* perform memory accesses on the location in question. Only actual byte reads/writes would count, and those are easy to deny if the size is 0.

Writing or reading 0 bytes *never does anything*, it's *not* a memory access.